File: volvo-fe-mdc/src/core/models/domain.model.ts
import CategoryModel from "./category.model";
import OperationModel from "./operation.model";

export interface DomainModel {
  id?: number;
  description?: string;
  inOut?: string;
  approvationNeeded?: boolean;
  visible?: boolean;
  categoryId?: number;
  category?: CategoryModel
  operationId?: number;
  operation?: OperationModel
  date?: Date
  isDetailedTransaction?: boolean;
}
export default DomainModel;

File: volvo-ms-ecash/Volvo.Ecash.Dto/Model/DomainModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Volvo.Ecash.Dto.Model
{
    [Table("domain")]
    public class DomainModel
    {
        [Column("id")]
        public int Id { get; set; }

        [Column("operation_id")]
        public int OperationId { get; set; }

        [Column("category_id")]
        public int CategoryId { get; set; }

        [Column("bank_account_id")]
        public int BankAccountId { get; set; }

        [Column("description")]
        public string Description { get; set; }

        [Column("aprovation_needed")]
        public bool ApprovationNeeded { get; set; }

        [Column("visible")]
        public bool Visible { get; set; }

        [Column("in_out")]
        public string InOut { get; set; }

        [Column("detailed_transaction")]
        public bool? IsDetailedTransaction { get; set; }

        [Column("report_order")]
        public int? ReportOrder { get; set; }

        public Operation Operation { get; set; }
        public Category Category { get; set; }

        public BankAccount BankAccount { get; set; }
    }
}
